src/Symfony/Component/Lock/TokenGenerator/RandomTokenGenerator.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\TokenGenerator;
+
+use Symfony\Component\Lock\TokenGeneratorInterface;
+
+/**
+ * RandomTokenGenerator is a TokenGeneratorInterface providing random string through the random_bytes function.
+ * kind of blocking storage.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class RandomTokenGenerator implements TokenGeneratorInterface
+{
+    private $entropy;
+
+    /**
+     * @param int $entropy The amount of entropy collected for each token (in bits)
+     */
+    public function __construct($entropy = 256)
+    {
+        $this->entropy = $entropy;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generateToken()
+    {
+        return base64_encode(random_bytes($this->entropy / 8));
```

Comment:
I suggest inlining this and remove `RandomTokenGenerator` and `TokenGeneratorInterface`: they are just internal details.